daffodil-core/src/test/scala/org/apache/daffodil/general/TestPrimitives.scala

```diff
@@ -137,6 +137,131 @@ class TestPrimitives {
+    TestUtils.assertEqualsXMLElements(expected, actual)
+  }
+
+  @Test def testSPDelimiter(): Unit = {
+    val sch = SchemaUtils.dfdlTestSchema(
+      <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>,
+
+      <dfdl:format ref="tns:GeneralFormat" representation="text" lengthUnits="bytes" encoding="US-ASCII" initiator="" separator="" terminator="" ignoreCase="no"/>,
+
+      <xs:element name="e1" dfdl:lengthKind="delimited">
+        <xs:complexType>
+          <xs:sequence dfdl:separator=",%SP;" dfdl:separatorPosition="infix">
+            <xs:element name="s1" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s2" type="xs:string" dfdl:lengthKind="delimited"/>
+          </xs:sequence>
+        </xs:complexType>
+      </xs:element>)
+    val (_, actual) = TestUtils.testString(sch, "SingleSpace, End")
+    val expected: Node = <e1><s1>SingleSpace</s1><s2>End</s2></e1>
+    TestUtils.assertEqualsXMLElements(expected, actual)
+  }
+
+  @Test def testSPPlusDelimiter(): Unit = {
+    val sch = SchemaUtils.dfdlTestSchema(
+      <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>,
+
+      <dfdl:format ref="tns:GeneralFormat" representation="text" lengthUnits="bytes" encoding="US-ASCII" initiator="" separator="" terminator="" ignoreCase="no"/>,
+
+      <xs:element name="e1" dfdl:lengthKind="delimited">
+        <xs:complexType>
+          <xs:sequence dfdl:separator=",%SP+;" dfdl:separatorPosition="infix">
+            <xs:element name="s1" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s2" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s3" type="xs:string" dfdl:lengthKind="delimited"/>
+          </xs:sequence>
+        </xs:complexType>
+      </xs:element>)
+    val (_, actual) = TestUtils.testString(sch, "SingleSpace, MultSpace,     End")
+    val expected: Node = <e1><s1>SingleSpace</s1><s2>MultSpace</s2><s3>End</s3></e1>
+    TestUtils.assertEqualsXMLElements(expected, actual)
+  }
+
+  @Test def testSPStarDelimiter(): Unit = {
+    val sch = SchemaUtils.dfdlTestSchema(
+      <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>,
+
+      <dfdl:format ref="tns:GeneralFormat" representation="text" lengthUnits="bytes" encoding="US-ASCII" initiator="" separator="" terminator="" ignoreCase="no"/>,
+
+      <xs:element name="e1" dfdl:lengthKind="delimited">
+        <xs:complexType>
+          <xs:sequence dfdl:separator=",%SP*;" dfdl:separatorPosition="infix">
+            <xs:element name="s1" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s2" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s3" type="xs:string" dfdl:lengthKind="delimited"/>
+          </xs:sequence>
+        </xs:complexType>
+      </xs:element>)
+    val (_, actual) = TestUtils.testString(sch, "None, SingleSpace,   MultSpace")
```

Comment:
Expected this data to be "None,SingleSpace, ..." to show that ,%SP*; would match zero spaces after the comma.